py-polars/polars/plugins.py

```diff
@@ -111,6 +112,7 @@ def _serialize_kwargs(kwargs: dict[str, Any] | None) -> bytes:
     return pickle.dumps(kwargs, protocol=5)
 
 
+@lru_cache(maxsize=16)
```

Comment:
a large number of pathlib objects can be heavy to cache: https://github.com/psf/black/issues/1950. so i think it's good to set a maxsize

I think it's very unlikely that anyone would be using more than 16 plugins in the same project anyway..

Reply:
Not in a hot loop at least.

Reply:
> a large number of pathlib objects can be heavy to cache: [psf/black#1950](https://github.com/psf/black/issues/1950). so i think it's good to set a maxsize
> 
> I think it's very unlikely that anyone would be using more than 16 plugins in the same project anyway..

Does 16 mean 16 different plugin functions or 16 distinct plugins?

Reply:
NVM, should be 16 plugin packages..